cli/cluster/cmd/publish.go

```diff
@@ -109,10 +116,45 @@ func replaceInstanceConfig(instance string, config *libcluster.Config,
 	}
 
 	cconfig, err = libcluster.ReplaceInstanceConfig(cconfig, instance, config)
-	if err != nil {
+	if err == nil {
+		// Instance is present in the configuration.
+		return libcluster.NewYamlConfigPublisher(publisher).Publish(cconfig.RawConfig)
+	}
+
+	var noInstanceError libcluster.NoInstanceError
+	if !errors.As(err, &noInstanceError) {
 		return fmt.Errorf("failed to replace an instance %q configuration "+
 			"in a cluster configuration: %w", instance, err)
 	}
+	if replicaset == "" {
+		return fmt.Errorf("failed to create an instance %q configuration: "+
+			"replicaset name is not specified", instance)
+	}
+	if group == "" {
+		// Try to determine a group.
+		for gname, cgroup := range cconfig.Groups {
+			found := false
+			for rname := range cgroup.Replicasets {
+				if rname == replicaset {
+					found = true
+					break
+				}
+			}
+			if found {
+				group = gname
+				break
+			}
+		}
```

Comment:
It could be a function in the libcluster.

Reply:
Extracted, thank you!